sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -1101,10 +1252,43 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
                                                         ProcessingStoppedReason reason,
                                                         CancellationToken cancellationToken)
         {
+            // TODO:  Implement this;  current code is just a place holder.
+            if (ActivePartitionProcessors.TryRemove(partitionId, out var processor))
+            {
+                processor.CancellationSource.Cancel();
+
+                try
+                {
+                    await processor.ProcessingTask.ConfigureAwait(false);
+                }
+                catch (TaskCanceledException)
+                {
+                    // nop
+                }
+                finally
+                {
+                    processor.CancellationSource.Dispose();
+                }
+            }
+
             await Task.Delay(1).ConfigureAwait(false);
-            throw new NotImplementedException();
         }
 
+        /// <summary>
+        ///   Performs the tasks needed invoke the <see cref="OnProcessingErrorAsync" /> method in the background,
+        ///   as it is intended to be a fire-and-forget operation.
+        /// </summary>
+        ///
+        /// <param name="exception">The exception that occurred during operation of the event processor.</param>
+        /// <param name="partition">The context of the partition associated with the error, if any; otherwise, <c>null</c>.</param>
+        /// <param name="operationDescription">A short textual description of the operation during which the exception occurred; intended to be informational only.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the processing.</param>
+        ///
+        private Task InvokeOnProcessingErrorAsync(Exception exception,
```

Comment:
This was needed in order to avoid a potential deadlock when using the synchronous `StopProcessing` call within the error handler, which is a case that we want to support.